### PR TITLE
Implemented java.lang.Integer

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by neacao** [comment]
Should to fix bug at test case first before create this PR

File: java/lang/Integer/IntegerTest.cpp [hunk 1]
```diff
@@ -308,16 +308,24 @@ TEST (JavaLang, IntegerLongValue) {
...
 
-
+    // Make a comparison between expected realResult and the real result
     Integer negativeInteger = -12345;
     expectedResult = -12345.00;
     realResult = negativeInteger.floatValue();
     ASSERT_TRUE(expectedResult == realResult);
+
+    // Make a comparison between the not expected realResult and the real result
+    Integer notExpectedInteger = -12345;
```
Comment (neacao) [inline java/lang/Integer/IntegerTest.cpp]: what is `notExpectedInteger` used for ?

File: java/lang/Integer/IntegerTest.cpp [hunk 2]
```diff
@@ -327,8 +335,15 @@ TEST (JavaLang, IntegerDoubleValue) {
...
     ASSERT_TRUE(expectedResult == realResult);
 
+    // Make a comparison between expected realResult and the real result
     Integer negativeInteger = -123456789;
     expectedResult = -123456789.0000000;
     realResult = negativeInteger.doubleValue();
     ASSERT_TRUE(expectedResult == realResult);
+
+    // Make a comparison between the not expected realResult and the real result
+    Integer notExpectedInteger = -12345;
```
Comment (neacao) [inline java/lang/Integer/IntegerTest.cpp]: Same as above

File: java/lang/Integer/IntegerTest.cpp (hunk 1 above)
Comment (ghost) [reply java/lang/Integer/IntegerTest.cpp]: for that code : ASSERT_FALSE(notExpectedResult == realResult);  
I want to test the case : the realResult and the expectedResult are different. So I call it notExpectedInteger.

File: java/lang/Integer/IntegerTest.cpp (hunk 2 above)
Comment (ghost) [reply java/lang/Integer/IntegerTest.cpp]: The same answer:

for that code : ASSERT_FALSE(notExpectedResult == realResult);  
I want to test the case : the realResult and the expectedResult are different. So I call it notExpectedInteger.

File: java/lang/Integer/IntegerTest.cpp (hunk 1 above)
Comment (neacao) [reply java/lang/Integer/IntegerTest.cpp]: I didn't see you use it below

**Review by neacao — APPROVED** [review]

**Pull request merged**